crates/spirv-builder/src/lib.rs

```diff
@@ -406,9 +406,7 @@ fn join_checking_for_separators(strings: Vec<impl Borrow<str>>, sep: &str) -> St
         let s = s.borrow();
         assert!(
             !s.contains(sep),
-            "{:?} may not contain separator {:?}",
-            s,
-            sep
+            "{}", "{s:?} may not contain separator {sep:?}"
```

Comment:
This seems wrong, it can only print the literal string you wrote, without formatting. I believe the issue is that we're still on Rust2018 (instead of Rust2021) and the lint might be misfiring?